lib/services/local/trust.go

```diff
@@ -293,6 +301,38 @@ func (s *CA) GetCertAuthorities(ctx context.Context, caType types.CertAuthType,
 	return cas, nil
 }
 
+// UpdateUserCARoleMap updates the role map of the userCA of the specified existing cluster.
+func (s *CA) UpdateUserCARoleMap(ctx context.Context, existingCluster, trustedCluster types.TrustedCluster) error {
+	existingCertAuthority, err := s.GetCertAuthority(ctx, types.CertAuthID{
+		Type:       types.UserCA,
+		DomainName: existingCluster.GetName(),
+	}, true)
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+		// CA may be deactivated.
```

Comment:
We already know if the CA is supposed to be active or not, so you can have the same flow for this whole thing (using the raw `CompareAndSwap` on the backend) just by choosing between `authorities` and `authorities/deactivated` as a prefix - duplicating code sucks, but so does the way we use multiple keys to hold the state of a trusted cluster, and I really don't want deactivated CAs to become an even bigger problem than they currently are.

Reply:
Something like this?